### DISCUSSION TITLE
How to add POIs during runtime?

### DISCUSSION THREAD

**Discussion by agent457** [mapsforge/mapsforge#1581] [Q&A] [answered]
(All of this is in an Android application)
I have a SQLite database I made of POIs.
Each POI has a unique ID, name, longitude, altitude, and description. (The POIs don't have tags because the tag that would describe them normally does not exist (they are rooms), if it's possible to make a custom one that would be great)
I want all these POIs to be added to the map the user uploads to the app on startup.
I know you can do this manually with the writer, but I want to do this on runtime.
How do I do that?

**Comment by devemux86** [comment, marked as answer]
Map files can only be created with map-writer.

You can display additional content as overlays, see the [OverlayMapViewer](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/OverlayMapViewer.java) example.

**Reply by agent457** [reply]
is there a way to add the map-writer to the the app and when the URI is loaded to run the command to add the POIs to it?

**Reply by devemux86** [reply]
@agent457 As explained [here](https://github.com/mapsforge/mapsforge/discussions/1568), map-writer is a desktop command line tool.
It is not intended to use inside apps, especially on Android.

Overlays is the proper way to display additional content on the map.
